jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -331,6 +333,45 @@ def _parse_spectrum1d(app, file_obj, data_label=None):
     app.add_data_to_viewer('spectrum-viewer', f"{data_label}[FLUX]")
 
 
+def _parse_ndarray(app, file_obj, data_label=None, data_type=None):
+    if data_label is None:
+        data_label = 'Unknown array'
+
+    if data_type is None:
+        data_type = 'flux'
+
+    flux = np.fliplr(np.rot90(np.moveaxis(file_obj, 0, 2), k=-1, axes=(0, 1)))
+    if not hasattr(flux, 'unit'):
```

Comment:
By happy accident, this also would take astropy Quantity because that inherits from ndarray, hence the need for this check. But I didn't do isinstance because Quantity has this weird magnitude support that is its own beast.